### PR TITLE
Obs id download

### PR DESCRIPTION
**Pull request by imbasimba**
Following or replacing: https://github.com/astropy/astroquery/pull/2034

Contains
1. Download by observation id or source name
3. http -> https

### CODE REVIEW

**Comment by keflavich** [comment]
This looks good at a glance (I did a shallow review).  Could you add a changelog entry?

**Comment by imbasimba** [comment]
Rebased and added changelog entry

File: astroquery/esasky/core.py
Comment (keflavich) [inline astroquery/esasky/core.py:799]: add a note in the parameter list that `position` is now optional

Comment (keflavich) [reply astroquery/esasky/core.py:799]: and maybe add a useful error message if a user fails to specify at least one of (position, observation_ids)

File: astroquery/esasky/core.py
Comment (keflavich) [inline astroquery/esasky/core.py:1371]: I think the preferred idiom is ``if row_limit != -1`

**Review by keflavich — APPROVED** [review]
I just have minor comments, this LGTM.  I'd suggest improving the docstring before merging; the other comment is just an optional style change.

**Comment by imbasimba** [comment]
Thanks for spotting those issues! They should all be fixed now.

File: docs/esasky/esasky.rst
Comment (bsipocz) [inline docs/esasky/esasky.rst:239]: Heading should have the same length as the text above, not sure why sphinx didn't complain in this case

```suggestion
Get the metadata of specific observations or sources
----------------------------------------------------

If you already know the observation ID's or source names of interest,
```

**Review by bsipocz — COMMENTED** [review]
I left some minor, mostly stylistic comments. Basically, I think it would be nicer to make kwargs only in more methods, noticed a duplicate of functionality, etc.

File: astroquery/esasky/core.py
Comment (bsipocz) [inline astroquery/esasky/core.py:110]: The first commit of adding TAP and ADQL functionality is indeed doesn't seem to be covered by tests. Could you please follow it up either here or in a new PR, if this is merged in the meantime?

File: astroquery/esasky/core.py
Comment (imbasimba) [reply astroquery/esasky/core.py:859]: It is for the case when the user has not specified one of the required arguments position and observation_ids. I believe this guard is required for this case, if I am not missing anything.

File: astroquery/esasky/core.py
Comment (imbasimba) [reply astroquery/esasky/core.py:1373]: Kindof. I've added an explanatory comment there plus a log in _build_id_query.

**Comment by imbasimba** [comment]
Hi @bsipocz, 

Great comments, as always! I believe I have addressed them all and I've also added a few tests for the TAP and ADQL functionality.

File: astroquery/esasky/core.py
Comment (bsipocz) [reply astroquery/esasky/core.py:859]: OK, let's get back to this separately as it's used all over the module. I still fail to see if an invalid entry is given, how would this safeguard. E.g. a nonsensical input it still passed through parse_coordinates, and it does nothing with None. So the difference is to give a slightly different error message:
```
In [8]: ESASky.get_images(position={'a':1})
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-8-e56760a8504d> in <module>
----> 1 ESASky.get_images(position={'a':1})

~/munka/devel/astroquery/astroquery/esasky/core.py in get_images(self, position, observation_ids, radius, missions, download_dir, cache)
    857         if position is None and observation_ids is None:
    858             raise ValueError("An input is required for either position or observation_ids.")
--> 859         sanitized_position = self._sanitize_input_position(position)
    860         sanitized_radius = self._sanitize_input_radius(radius)
    861         sanitized_missions = self._sanitize_input_mission(missions)

~/munka/devel/astroquery/astroquery/esasky/core.py in _sanitize_input_position(self, position)
   1062             return None
   1063         else:
-> 1064             raise ValueError("Position must be either a string or "
   1065                              "astropy.coordinates")
   1066 

ValueError: Position must be either a string or astropy.coordinates

In [9]: parse_coordinates({'a':1})
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-3665b880e983> in <module>
----> 1 parse_coordinates({'a':1})

~/munka/devel/astroquery/astroquery/utils/commons.py in parse_coordinates(coordinates)
    192             c = coord.SkyCoord(coordinates)
    193     else:
--> 194         raise TypeError("Argument cannot be parsed as a coordinate")
    195     return c
    196 

TypeError: Argument cannot be parsed as a coordinate
```

File: astroquery/esasky/core.py
Comment (bsipocz) [inline astroquery/esasky/core.py:895]:
```suggestion
    def get_spectra(self, *, position=None, observation_ids=None, radius=__ZERO_ARCMIN_STRING,
```

File: astroquery/esasky/core.py
Comment (bsipocz) [reply astroquery/esasky/core.py:1373]: In some other modules we raise an `InvalidQueryError` in cases like this, maybe that, or another Error type would make sense here, too and would be a quicker feedback for users that something is not right with their query. Anyway, this can be also a follow-up if you also think an error would make sense. 
(Either is fine by me at this point).

**Review by bsipocz — APPROVED** [review]

File: astroquery/esasky/core.py
Comment (bsipocz) [inline astroquery/esasky/core.py:727]:
```suggestion
    def get_maps(self, query_table_list, *, missions=__ALL_STRING,
```

**Comment by bsipocz** [comment]
My remaining comments are for a possible follow-up, and would rather go ahead and merge this once CI and remote tests pass to get it included in the imminent 0.4.3

**Comment by bsipocz** [comment]
Thanks @imbasimba!

File: astroquery/esasky/core.py
Comment (imbasimba) [reply astroquery/esasky/core.py:859]: Yeah, you are correct, the check is not necessary. I guess it only gives a different error message for something like ESASky.get_images()

Comment (bsipocz) [reply astroquery/esasky/core.py:859]: That case is handled in L858 😊

Comment (imbasimba) [reply astroquery/esasky/core.py:859]: :man_facepalming: I thought we were talking about L857-858... I'm an idiot. :man_facepalming:

Let's pretend the following is my first response to this comment:
Yes, _sanitize_input_position seem to have been a complete waste of space all these years. Good catch! I'll remove it at the next opportunity.

Comment (bsipocz) [reply astroquery/esasky/core.py:859]: Haha, this is on me, in sorry. I suppose I should stop reviewing coffee that hasn't changed in a PR 😊

**Pull request merged**
